src/api/app/controllers/attribute_controller.rb

```diff
@@ -298,4 +244,51 @@ def find_attribute_container
     params[:namespace] = name_parts[0]
     params[:name] = name_parts[1]
   end
+
+  private
+
+  def ensure_namespace
```

Comment:
This belongs to the routes. `namespace` is simply part of the route and if it is not there then this is a 404. No need to bother the controller with it.